src/udev/udev-node.c

```diff
@@ -329,6 +255,9 @@ static int stack_directory_open(const char *dirname) {
         if (fd < 0)
                 return -errno;
 
+        if (flock(fd, LOCK_EX) < 0)
```

Comment:
the dir is world-accessible, no? that kinda rules out flock() for this I fear. Because you can read a directory, then you can also lock it. That seriously limits the usefulness of flock(), but it is how it is...

Otherwise any arbitrary process on the system can make udev hang/DoS...

Reply:
but we can make the dir have access mode 0700, right? if so this would be OK to do I guess. or would we lose anything if we made that change? i.e. should have unpriv userspace have a good reason to read these dirs?

Reply:
other option, create a lock file inside the dir, and jst make the lock file 0700. Slightly complicates removing the dir when it is not used anymore though

Reply:
Now the directory is created with 0700, as creating a file in the directory makes the cleanup logic complicated.

Reply:
> the dir is world-accessible, no? that kinda rules out flock() for this I fear. Because you can read a directory, then you can also lock it. That seriously limits the usefulness of flock(), but it is how it is...
> 
> Otherwise any arbitrary process on the system can make udev hang/DoS...

Sorry but I don't get this. The lock file is created with 0600 perms.

How can an unpriv process could also lock it ?